torchdata/datapipes/iter/util/jsonparser.py

```diff
@@ -33,15 +33,13 @@ class JsonParserIterDataPipe(IterDataPipe[Tuple[str, Dict]]):
         [('1.json', ['foo', {'bar': ['baz', None, 1.0, 2]}]), ('2.json', {'__complex__': True, 'real': 1, 'imag': 2})]
     """
 
-    def __init__(self, source_datapipe: IterDataPipe[Tuple[str, IO]], **kwargs) -> None:
-        self.source_datapipe: IterDataPipe[Tuple[str, IO]] = source_datapipe
+    def __init__(self, source_datapipe: IterDataPipe, input_col=1, output_col=None, **kwargs) -> None:
+        super().__init__(source_datapipe, input_col=input_col, output_col=output_col)
         self.kwargs = kwargs
 
-    def __iter__(self) -> Iterator[Tuple[str, Dict]]:
-        for file_name, stream in self.source_datapipe:
+    def _map(self, stream: IO):
```

Comment:
Do you think we should change the type to `StreamWrapper` here?

Reply:
We are fine with any stream as input.